### PR TITLE
Ctosc 97 - migrate away from the in-tree configuration

### PR DESCRIPTION
**Pull request by bah-kvn**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -183,14 +183,14 @@ module "servers" {
...
-  block_device_mappings       = var.block_device_mappings
-  extra_block_device_mappings = var.extra_block_device_mappings
-  vpc_security_group_ids      = concat([aws_security_group.server.id, aws_security_group.cluster.id, module.cp_lb.security_group], var.extra_security_group_ids)
-  spot                        = var.spot
+  vpc_id                 = var.vpc_id
+  subnets                = var.subnets
+  ami                    = var.ami
+  instance_type          = var.instance_type
+  block_device_mappings  = var.block_device_mappings
+  extra_storage          = var.extra_storage
```
Comment (gosim-bah) [inline main.tf:191]: Unless there's a good reason to change this name, this should remain `extra_block_device_mappings` to match what upstream has.

https://github.com/rancherfederal/rke2-aws-tf/blob/c40e44a0b5ff561235d19514eee34ca7dee04575/modules/agent-nodepool/main.tf#L135

https://github.com/rancherfederal/rke2-aws-tf/blob/c40e44a0b5ff561235d19514eee34ca7dee04575/modules/nodepool/main.tf#L45

File: modules/nodepool/main.tf [hunk 2]
```diff
@@ -29,8 +29,10 @@ resource "aws_launch_template" "this" {
     security_groups             = concat(var.vpc_security_group_ids, [aws_security_group.this.id])
   }
 
+#FIXME: maybe device should be a variable? 
+#    device_name = lookup(var.block_device_mappings, "device_name", "/dev/sda1")
   block_device_mappings {
-    device_name = lookup(var.block_device_mappings, "device_name", "/dev/sda1")
+    device_name = lookup(var.block_device_mappings, "device_name", "/dev/xvda")
```
Comment (gosim-bah) [inline modules/nodepool/main.tf:35]: Why are we changing the mount location?

Comment (steven-terrana-bah) [reply modules/nodepool/main.tf:35]: apparently it differs per AMI being used, which isn't apparent by just looking at this PR. 

We should probably investigate using the [terraform `aws_ami` data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami) to query the AMI and then figure out what the device name is supposed to be. 

One of the output attributes is the `block_device_mappings` which has an attribute on each element for the `device_name`

**Review by steven-terrana-bah — CHANGES_REQUESTED** [review]

File: modules/nodepool/main.tf (hunk 2 above)
Comment (gosim-bah) [reply modules/nodepool/main.tf:35]: Huh, Today I learned. I could've sworn `sda` was specifically for root and/or block devices and that `xvd` was for extensions. I feel like the most recent time I saw that being used was for NVME SSDs.

Good to know.

File: main.tf (hunk 1 above)
Comment (bah-kvn) [reply main.tf:191]: Understood and fully agree. About half way through I was trying to get the names consistent between the two ( extra_block_device_mappings & extra storage ) and when deciding which was the "original" name - picked the wrong one - and ended up switching everything to extra_storage.  

There are a few variable names that are in upstream that could be better - extra_config_config comes to mind , and depending on perspective extra_block_device_mappings might also be a candidate to be clarified ( extra_block_device_mappings ? relative to what?   extra mappings for root vol? additional storage besides root ?  etc ).   I'm considering  opening  a pr for upstream for extra_config_config (at a minimum).

File: modules/nodepool/main.tf (hunk 2 above)
Comment (bah-kvn) [reply modules/nodepool/main.tf:35]: This gave us fits for a few days when the gp2/gp3 rules changed .  I think it depends on the bus type - for example /dev/sd is a scsi bus, /dev/hd is ide, xvd is xen paravirtual.  The next character is the order of the device on the bus ( a is the first device, b the second ...) and the trailing number is the partition number ( I think ).  Additionally , I think its possible that the drivers loaded might change the device names.  If you have the xen driver loaded - you should see the xvd device names, if not and you are using stock scsi - then you will see the sda.

I like the terraform aws_ami data source idea and think there are likely other places we might be able to do something similar to reduce the amount of stuff users have to provide - for example if I know the domain name I want to use - I can look up the hosted zone id .

**Comment by gosim-bah** [comment]
I mentioned this in a DM to Kevin, but the only thing I'm thinking about (just talking out loud) is whether the new mount points are going to break if we use _other_ AMIs / customer bring their own AMIs. This probably needs some more testing in the future but as for now this is good.

**Review by gosim-bah — APPROVED** [review]

**Pull request merged**